### PR TITLE
Wini Irarrazaval - Octo - Hotel

### PR DESCRIPTION
**Pull request by winirarrazaval**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? |  I had to decide if making a room class or not, I chose not to. I think that was a good decision, what I did not think about doing on time was a checking availability class. If I had done my code would be cleaner.
Describe a concept that you gained more clarity on as you worked on this assignment. | I have more clarity on how to use different classes and why are tests so important.
Describe a _nominal test_ that you wrote for this assignment. | When calling create a block method and instance of RoomBlock was created.
Describe an _edge case test_ that you wrote for this assignment. | Checking that when reserving in a block on stays shorter than the block, that room could be reserved again in the same reservation block.
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | Good, it gives me clarity on what I have to do.

### CODE REVIEW

File: lib/admin.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        the_block.reservations_in_this_block.each do |reservation|
+          range = (reservation.start_date..(reservation.end_date - 1))
+          if range === start_ || range === end_
+            unavailable_room_in_block << reservation.room_id
+          end
+        end
+        return unavailable_room_in_block
+      end
+
+      if block_id == nil
```
Comment (kariabancroft) [inline lib/admin.rb:65]: You should just use an `else` here

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Somewhat - The Reservation and Block have some duplication so i'd encourage you to think about how you might be able to leverage inheritance to DRY this up. Additionally, there is a lot of date range logic in the Admin class that you might be able to abstract out. 
Demonstrated loose coupling | Yes - The Admin class relies on a lot of the nitty gritty details of each individual class, so it'd be good to see how we might move some of this logic out
Methods demonstrate a good use of encapsulation, inputs and outputs |  Yes
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements |Yes

You should be committing using git more often with more meaningful commit messages.
